src/Model/City.php

```diff
@@ -60,58 +57,59 @@ public function __construct(array $raw, array $locales = ['en'])
     {
         parent::__construct($raw, $locales);
 
-        $this->city = new \GeoIp2\Record\City($this->get('city'), $locales);
-        $this->location = new \GeoIp2\Record\Location($this->get('location'));
-        $this->postal = new \GeoIp2\Record\Postal($this->get('postal'));
-
-        $this->createSubdivisions($raw, $locales);
-    }
+        $this->city = new \GeoIp2\Record\City($raw['city'] ?? [], $locales);
+        $this->location = new \GeoIp2\Record\Location($raw['location'] ?? []);
+        $this->postal = new \GeoIp2\Record\Postal($raw['postal'] ?? []);
 
-    private function createSubdivisions(array $raw, array $locales): void
-    {
         if (!isset($raw['subdivisions'])) {
+            $this->subdivisions = [];
+            $this->mostSpecificSubdivision =
+                    new \GeoIp2\Record\Subdivision([], $locales);
+
             return;
         }
 
+        $subdivisions = [];
         foreach ($raw['subdivisions'] as $sub) {
-            $this->subdivisions[] =
+            $subdivisions[] =
                 new \GeoIp2\Record\Subdivision($sub, $locales)
             ;
         }
+
+        // Not using end as we don't want to modify internal pointer.
```

Comment:
Would `array_key_last` help with this ?

Reply:
Yes, but that would just turn it to ` $subdivisions[\array_key_last($subdivisions)]`. I am not sure that is a big improvement. I am guessing it would be slower and might make people think it is an associative array as opposed to an indexed one.